syntax = "proto3";

import "google/protobuf/empty.proto";

service NodeMonitor {
    rpc GetStatus (google.protobuf.Empty) returns (NodeStatus) {
    }
    rpc GetAllNodeStatus (google.protobuf.Empty) returns (AllNodeStatus) {
    }
}

service NodeActorMonitor {
    rpc GetNodeActorStatus (google.protobuf.Empty) returns (ActorStatus) {
    }
}

message NodeStatus {
    double cpu_use_rate = 1;
    double mem_use_rate = 2;
    uint64 max_memory = 3;
    uint64 use_memory = 4;
    string ip = 5;
    string node_name = 6;
    string node_role = 7;
    string error_message = 8;
}

message AllNodeStatus {
    repeated NodeStatus node_status = 1;
}

message ActorInfo {
    uint64 id = 1;
    string name = 2;
    string description = 3;
}

message ActorStatus {
    repeated ActorInfo actor_infos = 1;
    int32 actor_worker = 2;
    string ip = 3;
    string error_message = 4;
}
